src/client/ui/viewcontrollers/EditorViewController.c

```diff
@@ -0,0 +1,329 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ */
+
+#include "ui_local.h"
+#include "client.h"
+
+#include "EditorViewController.h"
+
+/**
+ * @brief
+ */
+void addButton(View *view, const char *title, ActionFunction function, ident sender, ident data) {
+
+	Button *button = $(alloc(Button), initWithFrame, NULL, ControlStyleDefault);
+
+	$(button->title, setText, title);
+
+	$((Control *) button, addActionForEventType, SDL_MOUSEBUTTONUP, function, sender, data);
+
+	$(view, addSubview, (View *) button);
+
+	release(button);
+}
+
+#define INPUT_LABEL_WIDTH 140
+
+/**
+ * @brief Adds a new Label and the specified Control to the given View.
+ *
+ * @remarks This function releases the Control for convenience.
+ */
+void addInput(View *view, const char *label, Control *control) {
+
+	assert(view);
+	assert(control);
+
+	Input *input = $(alloc(Input), initWithFrame, NULL);
+	assert(input);
+
+	$(input, setControl, control);
+
+	$(input->label->text, setText, label);
+
+	input->label->view.autoresizingMask &= ~ViewAutoresizingContain;
+	input->label->view.frame.w = INPUT_LABEL_WIDTH;
+
+	$(view, addSubview, (View *) input);
+
+	release(input);
+}
+
+#define _Class _EditorViewController
+
+#pragma mark - Actions & Delegates
+
+/**
+ * @brief ActionFunction for the Save Button.
+ */
+static void saveAction(Control *control, const SDL_Event *event, ident sender, ident data) {
+
+	printf("Save!\n");
+}
+
+/**
+ * @brief SliderDelegate callback for changing bump.
+ */
+static void didSetBump(Slider *self) {
+
+	EditorViewController *this = (EditorViewController *) self;
+
+	if (!this->material) {
+		return;
+	}
+
+	this->material->cm->bump = (vec_t) this->bumpSlider->value;
+}
+
+/**
+ * @brief SliderDelegate callback for changing hardness.
+ */
+static void didSetHardness(Slider *self) {
+
+	EditorViewController *this = (EditorViewController *) self;
+
+	if (!this->material) {
+		return;
+	}
+
+	this->material->cm->hardness = (vec_t) this->hardnessSlider->value;
+}
+
+/**
+ * @brief SliderDelegate callback for changing specular.
+ */
+static void didSetSpecular(Slider *self) {
+
+	EditorViewController *this = (EditorViewController *) self;
+
+	if (!this->material) {
+		return;
+	}
+
+	this->material->cm->specular = (vec_t) this->specularSlider->value;
+}
+
+/**
+ * @brief SliderDelegate callback for changing parallax.
+ */
+static void didSetParallax(Slider *self) {
+
+	EditorViewController *this = (EditorViewController *) self;
+
+	if (!this->material) {
+		return;
+	}
+
+	this->material->cm->parallax = (vec_t) this->parallaxSlider->value;
+}
+
+#pragma mark - ViewController
+
+/**
+ * @see View::updateBindings(View *)
+ */
+static void updateBindings(View *self) {
+
+	EditorViewController *this = (EditorViewController *) self;
+
+	if (!this->bumpSlider) { // we probably haven't run loadView yet
```

Comment:
Perhaps a better way to handle this would be by overriding `respondToEvent` -- or maybe we should modify ViewController so that it also has some lifecycle management. I'll think on this a little today.